timemachine/fe/rbfe.py

```diff
@@ -408,9 +408,10 @@ def estimate_free_energy_given_initial_states(initial_states, protocol, temperat
     ukln_by_lambda_by_component = np.array(ukln_by_component_by_lambda).swapaxes(0, 1)
 
     def pair_overlap(u_kln):
-        assert u_kln.shape[:2] == (2, 2)
-        u_kn = np.concatenate(u_kln, axis=1)
-        N_k = u_kln.shape[2] * np.ones(u_kn.shape[0])
+        k, l, n = u_kln.shape
+        assert k == l == 2
+        u_kn = u_kln.reshape(k, -1)  # (k, l * n)
```

Comment:
Maybe comment -> assertion?
```suggestion
        u_kn = u_kln.reshape(k, -1)
        assert u_kn.shape == (k, l * n)
```

Reply:
Good call, added in [85f052f](https://github.com/proteneer/timemachine/pull/823/commits/85f052f684fe01cc569212f384ec8944b54d1c26)